lib/tlTimeline/TimelinePlayerPrivate.cpp

```diff
@@ -510,18 +554,33 @@ namespace tl
 
                         if (p->audioThread.convert)
                         {
-                            p->audioThread.buffer.push_back(p->audioThread.convert->convert(tmp));
+                            const auto convertedAudio = p->audioThread.convert->convert(tmp);
+                            p->audioThread.buffer.push_back(convertedAudio);
```

Comment:
I think it might be better to reverse the audio after it is converted, it will remove the need for the extra copy and also simplify the reverse function since it only needs to deal with a single audio stream.

Reply:
You cannot reverse the audio after it is conveted (that was my first attempt) as when you have dissolves, you'd reverse them too.

Reply:
One thing that I left for optimizing is that currently I copy all the audio data instead of just "size".

Reply:
Audio dissolves are not implemented yet, but I think they would happen in the ``audio::mix()`` function. That function takes multiple audio tracks and mixes them into a single output stream. How would reversing that output stream cause issues with the transitions?

Copying a bit of extra data is probably fine and could be done as a follow up PR.

Reply:
Think about it.  If you have audio frames 10-40 and the audio fade starts at frame 15 and you go and reverse that section the audio, the audio fade would happen at 40-5 (ie. 35, not 15 as when playing forwards).